### DISCUSSION TITLE
GetUnspentUTXOs(excludeUnconfirmed = true) returns unconfirmed UTXOs

### DISCUSSION THREAD

**Discussion by danielalexiuc** [btcpayserver/btcpayserver#4395] [Technical Support]
https://github.com/btcpayserver/btcpayserver/blob/5387a6287ea5b8e4aec416af72c325cd29604443/BTCPayServer/Services/Wallets/BTCPayWallet.cs#L342

I still seem to be getting unconfirmed utxos in my inputs when there are no confirmed ones available - even when excludeUnconfirmed=true. I believe this call is to NBitcoin? 

What does it do if there are no confirmed utxos, will it return unconfirmed ones or fail?

**Comment by NicolasDorier** [comment]
The code is here https://github.com/dgarage/NBXplorer/blob/master/NBXplorer.Client/Models/UTXOChanges.cs#L63

**Reply by NicolasDorier** [reply]
It doesn't seems to me it should return unconf.

**Reply by NicolasDorier** [reply]
You can probably debug step by step in visual studio as pdb should have links to github. (need to uncheck "Enable just my code" in VS)

**Comment by NicolasDorier** [comment]
@danielalexiuc note it doesn't return unconfirmed but already spent utxo!
